massa-bootstrap/src/establisher.rs

```diff
@@ -1,91 +1,35 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
-use async_trait::async_trait;
 use massa_time::MassaTime;
-use std::{io, net::SocketAddr};
+use std::{
+    io,
+    net::{SocketAddr, TcpListener, TcpStream},
+};
 
 /// duplex connection
-pub type Duplex = tokio::net::TcpStream;
+pub trait Duplex:
+// static because need to send between threads :(
+    'static  + Send + tokio::io::AsyncReadExt + tokio::io::AsyncWriteExt + std::marker::Unpin
+{
+}
```

Comment:
Requiring a static lifetime here is fine on the server side as it will stay up for the whole duration of the program. But this trait is also used on the client side which is only going to exist for one bootstrap duration, and `'static` objects will outlive it even though we have no need for it. One solution could be to use reference counters.

Reply:
tried playing around with that, and couldn't get rid of the static.even with use of Arc, but the 'static constraint can be moved to constraints on implementations that only apply in a server-context (not client)

can be found here: [f040213](https://github.com/massalabs/massa/pull/3685/commits/f040213525793787e28c85a81d743f234207c545)